earnings/database/Earning/dataGen.js

```diff
@@ -1,86 +1,219 @@
-    // Randomly insert a real stock tickerID from stockList companyData
-    if (Math.random() * 500 % 7 === 3) {
-        tickerID = companyData.ticker;
-    } else {
-        tickerID = tickerIDArr.join('');
+
+    // Zero-space the ticker id, currently stored as base 26 digits
+    for (let j = digits.length; j < tickerLength; j++) {
+      digits.unshift(0);
     }
 
-    if (allTickers.hasOwnProperty(tickerID)) {
-        return createTickerID();
-    } else {
-        return tickerID;
+    // Translate to real letters, array to string
+    for (let i = 0; i < digits.length; i++) {
+      arrayTicker.push(alphabet[digits[i]]);
     }
-}
+    let tickerId = arrayTicker.join('');
 
-const createCompanyName = () => {
-    faker.company.companyName().split(',').join(' ');
-}
+    // Check for duplicates
+    this.count += 1;
+    this.next += 1;
+    if (noDuplicateOriginal100.includes(tickerId)) {
+      const avoidDuplicate = this.turnCountIntoId(this.next);
+      return avoidDuplicate;
+    } else {
+      return tickerId;
+    } 
+  }
 
-const generator = () => {
-    // Populate all companies object with just one company
-    // const numberOfCompanies = 26*26*26*26*22;
-    const allCompanies = [];
-    const numberOfCompanies = 1;
-    for (let i = 0; i < numberOfCompanies; i++) {
-        allCompanies.push({
-            ticker: createTickerID(),
-            company: faker.company.companyName().split(',').join(''),
-        });
+  // Make or retrieve company names, via the original 100 stocks array
+  createCompanyName (num) {
+    if (num < 100) {
+      return originalStocksArr[num].company;
+    } else {
+      return faker.company.companyName().split(',').join(' ');
     }
+  }
 
-    // Make sample data from quarterly earnings per eachCompany
-    for (const eachCompany of allCompanies) {
-        let actualEarning = Math.random() * 7;
-        let estimatedEarning = actualEarning;
-        
-        let quarterNumber = 0;
-        for (const quarter of EPSdate) {
-            let range = Math.floor(Math.random() * 100);
-            range *= Math.floor(Math.random() * 2) === 1 ? 0.45 : -0.40;
-            actualEarning *= (1 + range / 100);
-            actualEarning = actualEarning.toFixed(2);
-
-            let estimateRange = Math.floor(Math.random() * 100);
-            estimateRange *= Math.floor(Math.random() * 2) === 1 ? 0.10 : -0.10;
-            estimatedEarning = actualEarning * (1 + estimateRange / 100);
-            estimatedEarning = estimatedEarning.toFixed(2);
-
-            sampleEarnings.push({
-                ticker: eachCompany.ticker,
-                company: eachCompany.company,
-                actualEarning: Number(actualEarning),
-                estimatedEarning: Number(estimatedEarning),
-                quarter,
-                quarterNumber,
-            });
-            quarterNumber += 1;
-        }
+  // Make seven quarterly earnings stock data at a time, per stock ticker; increment count
```

Comment:
This is awesome! I would be curious if you look @ top when you are running this, if the generator holds the full context in memory through closure, or if this is allowing JS to garbage collect unused memory between generator calls!